File: Roorkee-sales-project-main/backend/app/api/v1/routers/analytics.py
import logging

from fastapi import APIRouter, Depends, Query
from sqlalchemy.orm import Session

from app.db.session import get_db
from app.schemas.analytics import CustomerInsights, RegionPerformance
from app.schemas.dashboard import RevenueByState, SalesTrendPoint
from app.services import analytics_service, dashboard_service

logger = logging.getLogger(__name__)
router = APIRouter()


@router.get("/sales-trend", response_model=list[SalesTrendPoint])
def get_sales_trend(
    months: int = Query(12, ge=1, le=36), db: Session = Depends(get_db)
) -> list[SalesTrendPoint]:
    return dashboard_service.get_sales_trend(db, months=months)


@router.get("/revenue-by-state", response_model=list[RevenueByState])
def get_revenue_by_state(db: Session = Depends(get_db)) -> list[RevenueByState]:
    return dashboard_service.get_revenue_by_state(db)


@router.get("/region-performance", response_model=list[RegionPerformance])
def get_region_performance(db: Session = Depends(get_db)) -> list[RegionPerformance]:
    return analytics_service.get_region_performance(db)


@router.get("/customer-insights", response_model=CustomerInsights)
def get_customer_insights(db: Session = Depends(get_db)) -> CustomerInsights:
    return analytics_service.get_customer_insights(db)

File: Roorkee-sales-project-main/frontend/src/api/analytics.ts
import { apiClient } from './client'
import type { CustomerInsights, RegionPerformance, RevenueByState, SalesTrendPoint } from './types'

export const getAnalyticsSalesTrend = (months = 12) =>
  apiClient.get<SalesTrendPoint[]>('/analytics/sales-trend', { params: { months } }).then((res) => res.data)

export const getAnalyticsRevenueByState = () =>
  apiClient.get<RevenueByState[]>('/analytics/revenue-by-state').then((res) => res.data)

export const getRegionPerformance = () =>
  apiClient.get<RegionPerformance[]>('/analytics/region-performance').then((res) => res.data)

export const getCustomerInsights = () =>
  apiClient.get<CustomerInsights>('/analytics/customer-insights').then((res) => res.data)
